librz/include/rz_util/rz_print.h

```diff
@@ -137,20 +137,23 @@ RZ_API char *rz_print_hexpair(RzPrint *p, const char *str, int idx);
 RZ_API RzPrint *rz_print_new(void);
 RZ_API RzPrint *rz_print_free(RzPrint *p);
 RZ_API void rz_print_set_flags(RzPrint *p, int _flags);
-RZ_API void rz_print_addr(RzPrint *p, ut64 addr);
-RZ_API void rz_print_section(RzPrint *p, ut64 at);
+RZ_API RZ_OWN char *rz_print_addr_str(RZ_NULLABLE RzPrint *p, ut64 addr);
+RZ_API void rz_print_addr(RZ_NONNULL RzPrint *p, ut64 addr);
+RZ_API RZ_BORROW const char *rz_print_section_str(RZ_NULLABLE RzPrint *p, ut64 at);
 RZ_API void rz_print_hexii(RzPrint *p, ut64 addr, const ut8 *buf, int len, int step);
-RZ_API void rz_print_hexdump(RzPrint *p, ut64 addr, const ut8 *buf, int len, int base, int step, size_t zoomsz);
-RZ_API int rz_print_jsondump(RzPrint *p, const ut8 *buf, int len, int wordsize);
-RZ_API void rz_print_hexdiff(RzPrint *p, ut64 aa, const ut8 *a, ut64 ba, const ut8 *b, int len, int scndcol);
+RZ_API RZ_OWN char *rz_print_hexdump_str(RZ_NONNULL RzPrint *p, ut64 addr, RZ_NONNULL const ut8 *buf, int len, int base, int step, size_t zoomsz);
```

Comment:
I guess you are converting these functions into `_str` ones because you plan to re-use them in Cutter? How do you plan to do so?

I'm asking because I feel like these functions are very rizin-specific and I'm not sure how much of them is going to be useful outside of it (e.g. in cutter).

Reply:
@ret2libc this is how `px` commands are used in Cutter: https://github.com/rizinorg/cutter/blob/dev/src/core/Cutter.cpp#L4194